### PR TITLE
Strengthen exception specification for `complex`

### PR DESCRIPTION
**Pull request by frederick-vs-ja**
Conditional `noexcept` specifications make the strengthening only applied to `complex<float>`, `complex<double>`, and `complex<long double>`. Not sure whether unconditional `noexcept` would be better (should be OK since `complex<SomeOtherType>` has unspecified effects).

This PR is currently adding `noexcept` to some internal functions declared in `<ymath.h>` and some other functions called by them. I hope this is not too aggressive.

### CODE REVIEW

**Review by strega-nil-ms — APPROVED** [review]
This looks good to me!

File: tests/std/tests/Dev10_555491_complex_linker_errors/test.cpp [hunk 1]
```diff
@@ -188,6 +188,117 @@ void test_gh_2728() {
...
+    STATIC_ASSERT(noexcept(pow(f, c)));
+}
+
+template <class F1, class F2>
+void test_strengthened_exception_specification_2() {
+    STATIC_ASSERT(is_nothrow_constructible_v<complex<F1>, const complex<F2>&>);
+    STATIC_ASSERT(is_nothrow_constructible_v<complex<F1>, complex<F2>>);
+
+    STATIC_ASSERT(is_nothrow_assignable_v<complex<F1>, const complex<F2>&>);
+    STATIC_ASSERT(is_nothrow_assignable_v<complex<F1>, complex<F2>>);
```
Comment (StephanTLavavej) [inline tests/std/tests/Dev10_555491_complex_linker_errors/test.cpp:288]: LHS should use lvalue references here.

File: stl/src/xmath.hpp [hunk 2]
```diff
@@ -104,11 +104,11 @@ union _Dval { // pun floating type as integer array
 unsigned short* _Pmsw(double*);
 
 short _Dint(double*, short);
-short _Dnorm(_Dval*);
-short _Dscale(double*, long);
+short _Dnorm(_Dval*) noexcept;
+short _Dscale(double*, long) noexcept;
 short _Dunscale(short*, double*);
```
Comment (StephanTLavavej) [inline stl/src/xmath.hpp:109]: It's weird to mark some, but not all, of this internal machinery as `noexcept`. We should just get it all.

**Review by StephanTLavavej — COMMENTED** [review]
Thanks! I'll validate and push changes...

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
@strega-nil-ms FYI, I pushed minor fixes and moderate-size `noexcept` additions to `xmath.hpp` and its implementation files after you approved. 

Also, I noticed that `_FPoly` and `_LDnorm` were never defined, so I removed their declarations. I verified that the export surfaces of the release and debug DLLs are unchanged.

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
:mechanical_arm: :abacus: :teacher:
===

**Pull request merged**